common/common-data/src/commonMain/kotlin/io/github/vasilyrylov/archsample/common/data/repository/AuthorizedUserRepository.kt

```diff
@@ -0,0 +1,47 @@
+package io.github.vasilyrylov.archsample.common.data.repository
+
+import io.github.vasilyrylov.archsample.common.data.mapper.UserMapper
+import io.github.vasilyrylov.archsample.common.data.preferences.IPreferences
+import io.github.vasilyrylov.archsample.common.domain.interfaces.IAuthorizedUserRepository
+import io.github.vasilyrylov.archsample.common.domain.model.UserId
+import io.github.vasilyrylov.archsample.data.database.ArchSampleDatabase
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.IO
+import kotlinx.coroutines.withContext
+
+class AuthorizedUserRepository(
+    private val preferences: IPreferences,
+    private val database: ArchSampleDatabase,
+) : IAuthorizedUserRepository {
+    override suspend fun isUserAuthorized(): Boolean {
+        val user = withContext(Dispatchers.IO) {
```

Comment:
Room should switch dispatcher under the hood. Please check it

Reply:
Short research showed me that switching does not for all cases, to be pragmatic, i prefer to set dispatcher explicitly.

See ExecutorCoroutineDispatcherImpl

```
    override fun dispatch(context: CoroutineContext, block: Runnable) {
        try {
            executor.execute(wrapTask(block))
        } catch (e: RejectedExecutionException) {
            unTrackTask()
            cancelJobOnRejection(context, e)
            Dispatchers.IO.dispatch(context, block)
        }
    }
```

Reply:
Then it is better to pass the dispatcher in the constructor parameters, so that in case of writing tests it can be redefined. But here it depends on further development. I do not insist